File: wheels/2.实用功能封装/2.2handlePhotos/ZHHandlePhotos/ViewController.m
//
//  ViewController.m
//  ZHHandlePhotos
//
//  Created by Hayder on 16/7/8.
//  Copyright © 2016年 wangzhenhai. All rights reserved.
//

#import "ViewController.h"
#import "ZHImageVIew.h"
#import "ZHImagesView.h"

@interface ViewController ()<ZHImagesViewDelegate>

@property (weak, nonatomic) IBOutlet UIImageView *imageView1;

@property (weak, nonatomic) IBOutlet UIImageView *imageView2;

@property (weak, nonatomic) IBOutlet UIImageView *imageView3;

@property (weak, nonatomic) IBOutlet UIImageView *imageView4;

@property (weak, nonatomic) IBOutlet UIImageView *imageView5;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.view.backgroundColor = [UIColor whiteColor];
    
    NSMutableArray *images = [NSMutableArray array];
    
    UIImage *image1 = [UIImage imageNamed:@"1.jpg"];
    [images addObject:image1];
    UIImage *image2 = [UIImage imageNamed:@"2.jpeg"];
    [images addObject:image2];
    UIImage *image3 = [UIImage imageNamed:@"3.jpg"];
    [images addObject:image3];
    UIImage *image4 = [UIImage imageNamed:@"4.jpg"];
    [images addObject:image4];
    UIImage *image5 = [UIImage imageNamed:@"5.jpeg"];
    [images addObject:image5];
    
    
    //1.创建对象
    ZHImagesView *imagesView = [ZHImagesView imagesView];
    //2.设置截取的比例 高/宽
    imagesView.snipScale = 1;
    //3.设置要处理的图片数组
    imagesView.images = images;
    //4.设置代理
    imagesView.imagesDelegate = self;
    //5.添加到视图上
    [self.view addSubview:imagesView];
    
}

- (void)imagesView:(ZHImagesView *)imagesView handleImagesResult:(NSArray *)resultImages
{
    NSLog(@"%@",resultImages);
    
    self.imageView1.image = resultImages[0];
    self.imageView2.image = resultImages[1];
    self.imageView3.image = resultImages[2];
    self.imageView4.image = resultImages[3];
    self.imageView5.image = resultImages[4];
}

@end

File: Demo/Swift/1.Swift3.0-变量和常量/1.Swift3.0-变量和常量/ViewController.swift
//
//  ViewController.swift
//  1.Swift3.0-变量和常量
//
//  Created by Hayder on 16/9/20.
//  Copyright © 2016年 wangzhenhai. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        Dome1()
        
        //关于 var  和 let 的选择
        //不可变的会更加安全，开发的时候，通常会先使用let，在需要变化的时候，再改成var
        
        
        var x = 10
        let y = 20
        
        let v = UIView()
        
        //仅仅修改的是V的属性，并没有修改v的指针地址
        v.backgroundColor = #colorLiteral(red: 0.9254902005, green: 0.2352941185, blue: 0.1019607857, alpha: 1)
        
        view.addSubview(v)
        
    }

    //1.定义变量 var，定义之后，可以修改
    // 常量 let, 定义之后，不能修改
    
    //2.自动推导，变量/常量的类型会根据右侧的代码执行结果，推倒对应的类型。
    //热键：option + click
    
    //整数:Int （oc中 NSInteger 类似）
    
    //3.在Swift中对类型要求异常严格
    //***（重要）任何不同类型的数据之间，不允许直接运算
    //不会做默认的隐式的转换，所有的类型确定，都要由程序员负责。
    
    //4.Swift中，不存在基本数据类型，都是结构体
    func Dome1() {
        
        let x = 10
        let y = 10.5
        
        //1. 将y装换成整数
        //OC中的写法（int）y - > 类型墙砖
        //Swift 中 Int() 结构体的构造函数
        print(x + Int(y))
        
        //2.将 x 转换成Double
        print(Double(x) + y)
        
    }
    func demo() {
        
 
        var x = 10
        x = 20
        
        //小数:Double 精度高
        let y = 10.5
        
        let view = UIView()
        
        print(x)
        print(y)
        print(view)
    }

    func Demo2() {
        
        //如果需要制定变量/常量的类型，也可以直接使用let x : 类型 = 值
        // 提示:在Swift 开发中，极少使用直接制定类型，通常都是自动推导
        let x : Double = 10
        let y = 10.5
        
        print(x + y)
        
        
    }

}

